### PR TITLE
Common make build, Arista containers, common log format, cleanup

### PR DESCRIPTION
**Pull request by GouthamBanala**
Breaking IPTCRC previous PR by including only common changes ------------
Common make build : Just get into container and build with 'make all' .
Added build variables for cross compilation
Makefile changes
Aggregated coverage reporting  
Added Arista build containers for 32 and 64 bit 
Corrected the log format for Arista
Renamed the folders and file appropriately

### CODE REVIEW

File: docker-build/lom-slave-bullseye/lom_start.sh
Comment (GouthamBanala) [inline docker-build/lom-slave-bullseye/lom_start.sh:1]: Goutham -- Check this and delete

File: docker-build/arista/arista64/lom-slave-bullseye/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        patchelf \
+        tree \
+        git \
+        libtool \
+        vim \
+        jq \
+        curl \
+        unzip \
+        zip  \
+uuid-dev \
```
Comment (GouthamBanala) [inline docker-build/arista/arista64/lom-slave-bullseye/Dockerfile]: formatting space

File: docker-build/arista/arista32/slave_docker_run.sh
Comment (GouthamBanala) [inline docker-build/arista/arista32/slave_docker_run.sh:1]: Put Readme  -- gnmi server issues

File: lom/Makefile [hunk 2]
```diff
@@ -235,118 +369,172 @@ $(ENGINE_TEST): $(SRC_FILES) $(TEST_FILES)
...
+		-gcflags=all=-l lom/build/src/gnmi/gnmi_server/server lom/build/src/gnmi/lom_data_clients || (echo "gNMIServerTest failed $$?"; exit 1); \
+	${GO} tool cover -html=$(TEST_DIR)/gNMIServerCoverprofile.out -o $(TEST_DIR)/gNMIServerCoverprofile.html; \
+        if [ -f "$(TEST_DIR)/gNMIServerCoverprofile.html" ]; then \
 		$(BUILD_DIR)/extract_file_coverage.sh $(TEST_DIR)/gNMIServerCoverprofile.html; \
+fi; \
+        touch $@; \
+        echo -e "Successfully completed gNMIServer Test\n\n"; \
+    fi
+
+#installer for Arista
```
Comment (GouthamBanala) [inline lom/Makefile:497]: don't review for now -- arista specific

File: lom/build-deprecated.sh
Comment (GouthamBanala) [inline lom/build-deprecated.sh:1]: remove it

File: lom/src/engine/engine/engine.go [hunk 3]
```diff
@@ -196,22 +199,41 @@ func EngineStartup(path string) (*engine_t, error) {
...
+    /* Parse args for path and mode*/
     p := ""
+    mode := ""
+var syslogLevelFlag int
     flags := flag.NewFlagSet(progname, flag.ContinueOnError)
     var buf bytes.Buffer
     flags.SetOutput(&buf)
 
     flags.StringVar(&p, "path", "", "Config files path")
+    flags.StringVar(&mode, "mode", "", "Mode of operation. Choice: PROD, test")
```
Comment (GouthamBanala) [inline lom/src/engine/engine/engine.go:212]: Add it in code as comments todo  :::  globals.conf.json  must have syslog level, do not pass as argument

Comment (GouthamBanala) [inline lom/src/engine/engine/engine.go:212]: Add as code comments :: must be in global level on all platforms

File: lom/src/engine/engine/engine.go [hunk 4]
```diff
@@ -196,22 +199,41 @@ func EngineStartup(path string) (*engine_t, error) {
...
+    flags.IntVar(&syslogLevelFlag, "syslog_level", 6, "Syslog level")
 
     err := flags.Parse(args)
     if err != nil {
         return nil, LogError("Failed to parse (%v); details(%s)", args, buf.String())
     }
+
+    if mode == "PROD" {
+        SetLoMRunMode(LoMRunMode_Prod)
+        InitSyslogWriter(p)
```
Comment (GouthamBanala) [inline lom/src/engine/engine/engine.go:224]: Renuka : How to differentiate UT vs non UT

File: lom/src/lib/lomcommon/helper.go [hunk 5]
```diff
@@ -35,15 +36,55 @@ func RunPanic(m string) {
 var DoPanic = RunPanic
 
 func init() {
+    InitSyslogWriter("")
+}
+
+func InitSyslogWriter(path string) {
```
Comment (GouthamBanala) [inline lom/src/lib/lomcommon/helper.go:43]: make path as global const and load config from init(). Add in code comment as TODO

**Review by renukamanavalan — DISMISSED** [review]

File: lom/src/plugins/vendors/sonic/plugin/linkcrc/link_crc_detection.go [hunk 6]
```diff
@@ -61,11 +61,11 @@ func NewLinkCRCDetectionPlugin(...interface{}) plugins_common.Plugin {
 func init() {
     // Register the plugin with plugin manager
     plugins_common.RegisterPlugin(link_crc_plugin_name, NewLinkCRCDetectionPlugin)
-}
+    }
 
 /* Inheritied from Plugin */
 func (linkCrcDetectionPlugin *LinkCRCDetectionPlugin) Init(actionConfig *lomcommon.ActionCfg_t) error {
-    lomcommon.LogInfo("Started Init() for (%s)", "link_crc")
+    lomcommon.LogInfo("Started Init() for (%s)", "link_crc_detection")
```
Comment (prithviraghavtm) [inline lom/src/plugins/vendors/sonic/plugin/linkcrc/link_crc_detection.go]: can use the variable above.

File: lom/src/plugins/vendors/sonic/plugin/linkcrc/link_crc_detection.go [hunk 7]
```diff
@@ -102,7 +102,7 @@ func (linkCrcDetectionPlugin *LinkCRCDetectionPlugin) Init(actionConfig *lomcomm
         lomcommon.LogError(fmt.Sprintf(link_crc_prefix+"Plugin initialization failed. (%s), err: (%v)", actionConfig.Name, err))
         return err
     }
-    lomcommon.LogInfo("Successfully Init() for (%s)", "link_crc")
+    lomcommon.LogInfo("Successfully Init() for (%s)", "link_crc_detection")
```
Comment (prithviraghavtm) [inline lom/src/plugins/vendors/sonic/plugin/linkcrc/link_crc_detection.go]: Same here: can use the local const variable.

**Review by prithviraghavtm — DISMISSED** [review]
Changes look good. Approving with few comments.

File: docker-build/arista/arista32/slave_docker_run.sh
Comment (GouthamBanala) [reply docker-build/arista/arista32/slave_docker_run.sh:1]: added REDAME with note

File: docker-build/arista/arista64/lom-slave-bullseye/Dockerfile (hunk 1 above)
Comment (GouthamBanala) [reply docker-build/arista/arista64/lom-slave-bullseye/Dockerfile]: fixed

File: docker-build/lom-slave-bullseye/lom_start.sh
Comment (GouthamBanala) [reply docker-build/lom-slave-bullseye/lom_start.sh:1]: Its deleted

File: lom/build-deprecated.sh
Comment (GouthamBanala) [reply lom/build-deprecated.sh:1]: done

File: lom/src/engine/engine/engine.go (hunk 3 above)
Comment (GouthamBanala) [reply lom/src/engine/engine/engine.go:212]: added

Comment (GouthamBanala) [reply lom/src/engine/engine/engine.go:212]: added

File: lom/src/engine/engine/engine.go (hunk 4 above)
Comment (GouthamBanala) [reply lom/src/engine/engine/engine.go:224]: added as To-DO

File: lom/src/lib/lomcommon/helper.go (hunk 5 above)
Comment (GouthamBanala) [reply lom/src/lib/lomcommon/helper.go:43]: addd to-do

File: lom/src/plugins/vendors/sonic/plugin/linkcrc/link_crc_detection.go (hunk 6 above)
Comment (GouthamBanala) [reply lom/src/plugins/vendors/sonic/plugin/linkcrc/link_crc_detection.go]: used it

File: lom/src/plugins/vendors/sonic/plugin/linkcrc/link_crc_detection.go (hunk 7 above)
Comment (GouthamBanala) [reply lom/src/plugins/vendors/sonic/plugin/linkcrc/link_crc_detection.go]: used the global const

**Review by prithviraghavtm — APPROVED** [review]

**Review by renukamanavalan — APPROVED** [review]

**Pull request merged**